applications/services/storage/storages/storage_ext.c

```diff
@@ -739,6 +740,13 @@ FS_Error storage_process_virtual_format(StorageData* storage) {
     SDData* sd_data = storage->data;
     uint8_t* work = malloc(_MAX_SS);
     SDError error = f_mkfs(sd_data->path, FM_ANY, 0, work, _MAX_SS);
+    storage_process_virtual_mount(storage);
+    const char* path = storage_file_get_path(mnt_image, mnt_image_storage);
+    char* label = basename(path);
+    int len = strlen(label);
+    label[len - 4] = '\0'; // truncate the .img extension
```

Comment:
This is effectively editing the memory of the furi string directly, which should never be done. You can malloc a smaller buffer of the appropriate size to get just filename without extension, and memcpy into it, then set the null terminator on that copied buffer

Reply:
Thanks for the feedback, will modify!

Reply:
Or use strlcpy with the actual buffer size (including the null terminator), which will figure it out appropriately

Reply:
Changes submitted. Thanks for the guidance, appreciate the feedback!